src/ensembl/src/content/app/custom-download/containers/header/helper.ts

```diff
@@ -0,0 +1,42 @@
+import config from 'config';
+
+import { SelectedAttribute } from 'src/content/app/custom-download/types/Attributes';
+
+export const fetchCustomDownloadResults = (
+  downloadType: string,
+  attributes: SelectedAttribute[],
+  filters: any
+) => {
+  let endpoint = config.genesearchAPIEndpoint + '/genes/fetch?query=';
+
+  let endpointFields = '';
+  attributes.forEach((attribute: SelectedAttribute) => {
+    endpointFields += attribute[2] + ',';
+  });
+
+  const endpointFilters: any = {
+    genome: 'homo_sapiens'
+  };
+
+  Object.keys(filters).forEach((filter: string) => {
+    endpointFilters[filter] = filters[filter];
+  });
+
+  endpoint =
+    endpoint +
+    JSON.stringify(endpointFilters) +
+    '&fields=' +
+    endpointFields +
+    '&sort=id&array=true&accept=' +
+    downloadType;
+  try {
+    setTimeout(() => {
+      const response = {
+        file: endpoint
+      };
+      window.open(response.file);
+    }, 100);
```

Comment:
- Why is this wrapped in try-catch?
- Why the setTimeout?
- What’s the purpose of the `response` object?

Reply:
Forgot to clean these up. I was initially trying to use a fake endpoint for the downloads but then I changed it to use the genesearch API itself.